Engine/ac/sys_events.cpp

```diff
@@ -678,7 +737,11 @@ static void sync_sys_mouse_pos()
 
 static void on_sdl_touch_down(const SDL_TouchFingerEvent &event)
 {
-    touch.fingers_down |= 1 << event.fingerId;
+    touch.fingers.push(event.fingerId);
+    int finger_index = touch.fingers.get_index(event.fingerId);
```

Comment:
If you make `push` return a new index, then you can save on 1 function call here, that is
```int finger_index = touch.fingers.push(event.fingerId);```

Reply:
If I do this, then I don't think I should do the iterator thing you suggested inside the push call, because then I need to calculate a distance to extract the index from the iterator.

Reply:
I was thinking about this:
```
    int push(SDL_FingerID fingerId)
    {
        if (contains(fingerId))
            return NO_ID;

        auto it = std::find(_fingers.begin(), _fingers.end(), NO_ID);
        if (it == _fingers.end()) return NO_ID;
        *it = fingerId;
        return it - _fingers.begin();
    };
```

Reply:
ok, so, imagining if two touch down occurred in sequence, the idea here is to dismiss one of them

Reply:
>  imagining if two touch down occurred in sequence, the idea here is to dismiss the second

Could you elaborate, I don't quite understand this?

Reply:
well, theoretically, that contain clause is there as a safety, but we should not need it. But in case the system misbehaves and we get two touch down before an up, using the same fingerId, then the second touch down is ignored. I think this is the correct behavior btw, so I agree with your code change.